Source/WTF/wtf/malloc_heap_breakdown/main.cpp

```diff
@@ -94,8 +94,10 @@ class MallocZoneHeapManager {
             zoneFree(zone, memory);
             return nullptr;
         }
-        m_zoneAllocations[zone][memory] = size;
-        return realloc(memory, size);
+        m_zoneAllocations[zone].erase(memory);
+        void* ptr = realloc(memory, size);
```

Comment:
On failure, `realloc` returns `NULL`, but the old pointer is still valid.
Here the code needs to do the test, to keep the tracking data for the old pointer, and to avoid tracking for `NULL`.